pkg/cosign/verify.go

```diff
@@ -647,23 +622,27 @@ func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 		if err != nil {
 			return bundleVerified, err
 		}
-		// If the chain annotation is not present or there is only a root
-		if chain == nil || len(chain) <= 1 {
-			co.IntermediateCerts = nil
-		} else if co.IntermediateCerts == nil {
-			// If the intermediate certs have not been loaded in by TUF
-			pool := x509.NewCertPool()
-			for _, cert := range chain[:len(chain)-1] {
-				pool.AddCert(cert)
+		// If the there is only a root. If there is no chain annotation present, preserve the
```

Comment:
I had to change this logic to make this work for blobs, where we are using the TUF intermediates/roots or supplying the whole chain via the CLI. I'll revisit this @haydentherapper